cub/util_device.cuh

```diff
@@ -348,6 +366,19 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
 
         result = cudaFuncGetAttributes(&empty_kernel_attrs,
                                        reinterpret_cast<void*>(empty_kernel));
+        // Any failure means either no GPU is available or the provided device
+        // binary does not match the generated function code
+        if (result != cudaSuccess) {
+            int sm_version;
+            int device;
+            if (!CubDebug(cudaGetDevice(&device))) {
+                if (!CubDebug(SmVersionUncached(sm_version, device))) {
+                    printf("Incompatible GPU: you are trying to run this program on sm_%d%d, "
+                            "different from the one that it was compiled for\n",
```

Comment:
Optional: we could report actual architectures we were compiled against. We are using the list in any case for our namespace generation:
```cpp
#define CUB_DETAIL_MAGIC_NS_BEGIN inline namespace CUB_DETAIL_MAGIC_NS_NAME(CUB_VERSION, NV_TARGET_SM_INTEGER_LIST) {
#else // not defined(_NVHPC_CUDA)
#define CUB_DETAIL_MAGIC_NS_BEGIN inline namespace CUB_DETAIL_MAGIC_NS_NAME(CUB_VERSION, __CUDA_ARCH_LIST__) {
```
We could probably concatenate appropriate lists and stringify the result as a follow up at some point.